src/middlewares/auth.js

```diff
@@ -1,59 +1,61 @@
-/* eslint-disable prefer-destructuring */
+import { config } from 'dotenv';
 import jwt from 'jsonwebtoken';
 
-const auth = {
-  generateToken(user) {
+config();
+
+export default {
+
+  generateToken(newUser) {
+    const {
+      id, first_name, last_name, email, role
+    } = newUser;
     return jwt.sign({
-      id: user.id,
-      role: user.role,
-    }, process.env.SECRET, { expiresIn: '7d' });
+      id,
+      first_name,
+      last_name,
+      email,
+      role
+    }, process.env.JWT_SECRET, { expiresIn: '7d' });
   },
 
-  verifyToken(token) {
-    let decoded = {};
-    try {
-      decoded.payload = jwt.verify(token, process.env.SECRET);
-    } catch (error) {
-      decoded = { error: error.message };
+  checkToken: (req, res, next) => {
+    const { authorization } = req.headers;
+    let token;
+
+
+    if (typeof authorization !== 'undefined' && authorization.includes('Bearer')) {
+      token = authorization.replace('Bearer ', '');
+    } else {
+      token = req.body.token;
+    }
+
+    if (!token) {
+      return res.status(403).json({
+        status: 'error',
+        error: 'Forbidden: You must be logged in to proceed',
+      });
     }
-    return decoded;
-  },
 
-  async verifyUserToken(req, res, next) {
     try {
-      let token;
-
-      if (req.headers.authorization) {
-        token = req.headers.authorization.split(' ')[1];
-      } else if (req.headers['x-access-token']) {
-        token = req.headers['x-access-token'];
-      } else if (req.headers.token) {
-        token = req.headers.token;
-      }
-
-      const decoded = auth.verifyToken(token);
-
-      if (!token) {
-        return res.status(401).json({
-          status: 'error',
-          error: 'No token provided.',
-        });
-      }
-
-      if (decoded.error) {
-        return res.status(401).json({
-          status: 'error',
-          error: 'Failed to authenticate token.',
-        });
-      }
+      req.user = jwt.decode(token, process.env.JWT_SECRET);
+      const {
+        user_id, role, first_name, last_name, email,
+      } = req.user;
+
+      /* pipe the token details into the request body */
+      req.body.user_id = user_id;
```

Comment:
I don;t think appending these values individually is a very good idea. This might conflict with future requests. You might want to leave it as `req.user` or `req,body.user`

Reply:
Appending token data back into the request body, as said in my previous comment, will prevent redundant requests in future. It will NEVER conflict with any request since at the end of the application lifecycle, there is always a RETURN statement which terminates that particular thread. If a new request is to be made, the application has to start a new lifecycle.

Reply:
To set user role, `role` (newRole) and `email` of the user to update are sent via the `req.body`. By appending `role` and `email` from the decoded `jwt`, you will overwrite the data being sent.

Reply:
Please come again. I couldn't really decipher what you mean by `overwriting data`.

Reply:
If you have time, we could continue this on zoom